server/src/main/java/io/deephaven/server/session/SessionState.java

```diff
@@ -801,20 +814,31 @@ private synchronized void setState(final ExportNotification.State state) {
 
                     errorHandler.onError(state, errorId, toReport, dependentHandle);
                 } catch (final Exception err) {
-                    log.error().append("Unexpected error while reporting state failure: ").append(err).endl();
+                    log.error().append("Unexpected error while reporting failure: ").append(err).endl();
                 }
             }
 
-            if (state == ExportNotification.State.EXPORTED || isExportStateTerminal(state)) {
+            final boolean isNowExported = state == ExportNotification.State.EXPORTED;
```

Comment:
1. I wonder if we should have more validation of state transitions, e.g. can't go from exported to exported.
2. Is `RELEASED` terminal? Not in the `isExportStateTerminal` impl.